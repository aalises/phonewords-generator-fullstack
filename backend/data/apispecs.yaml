basePath: "/api/v1"
schemes:
  - http
info:
  description: "Simple API to convert phone numbers to phone words. The base URL for retrieving the resources is /api/v1"
  title: "PhoneWords API"
  version: 1.0.0
externalDocs:
  description: "Project repository"
  url: "https://github.com/aalises/phonewords-generator-fullstack"  
produces: 
    - "application/json"    
paths:
  "/phonewords/<number>":
    get:
      summary: Computes all the phonewords associated with a number
      tags: 
        - resources
      produces:
        - application/json  
      description: Computes all the phonewords associated with a number
      responses: 
          "200":
            description: "Phone word response with a list of all possible phonewords, success flag and error message if applicable"
            schema:
              $ref: '#/definitions/Phonewords'
          default:
            description: "Unexpected Error"

definitions:
  Phonewords:
    required:
      - success
      - phonewords
      - error
    properties:
      success:
        type: boolean
        description: Success flag of the call
      phonewords:
        type: array
        items:
          type: string
        description: List with all the possible phonewords
      error:
        type: string
        description: Error message       
          
swagger: '3.0'