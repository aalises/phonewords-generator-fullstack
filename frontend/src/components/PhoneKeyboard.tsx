import React from 'react';
import styled from "styled-components";

//Orbit imports
import Heading from "@kiwicom/orbit-components/lib/Heading";
import Button from "@kiwicom/orbit-components/lib/Button";
import Card from "@kiwicom/orbit-components/lib/Card";
import CardSection from "@kiwicom/orbit-components/lib/Card/CardSection";
import CardSectionHeader from "@kiwicom/orbit-components/lib/Card/CardSection/CardSectionHeader";
import CardSectionContent from "@kiwicom/orbit-components/lib/Card/CardSection/CardSectionContent";

const numbers = ['1','2','3', '4','5','6', '7','8','9', 'Back','0','Clear'];


const handleClick = (el : string, setValues : function, resetForm: function) : void => {
  if(el === 'Clear'){
    resetForm()
  }else{
    setValues(el, el === "Back" ? true : false);
  }
}

interface PhoneKeyboardProps {
  setValues: (string, boolean) => void,
  resetForm: void => void
}

const KeyboardPhoneWrapper = styled.div`
  margin: 1rem 0rem
`

const Grid = styled.div`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 10px;
  width: 80%
`

export default ({setValues, resetForm} : PhoneKeyboardProps) => {
    return (
      <KeyboardPhoneWrapper>
        <Card>
          <CardSection expandable dataTest="keyboard-phone-card">
            <CardSectionHeader>
              <Heading type="title3" element="h3">Show/Hide Phone Keyboard </Heading>
            </CardSectionHeader>
            <CardSectionContent>
              <Grid>
              {numbers.map(el => 
                <Button type="secondary" dataTest={`keyboard-phone-element-${el}`} onClick={() => handleClick(el, setValues, resetForm)} key={el}>{el}</Button>
              )}
              </Grid>
            </CardSectionContent>
          </CardSection>
        </PhoneCard>
      </KeyboardPhoneWrapper>
    )
  }
}
