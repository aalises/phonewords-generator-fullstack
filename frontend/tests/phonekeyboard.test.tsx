import * as React from 'react';
import { shallow } from 'enzyme';
import PhoneKeyboard from '../src/components/PhoneKeyboard';
import data from "./data";

describe("Phoneword Keyboard Phone Test Suite", () => {

  it('Renders the phone keyboard correctly', () => {
    const phone =  shallow(<PhoneKeyboard handleClick={_ => _} />);
    expect(phone).toMatchSnapshot();
  });
});