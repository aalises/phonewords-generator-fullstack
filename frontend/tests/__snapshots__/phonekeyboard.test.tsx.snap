// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phoneword Keyboard Phone Test Suite Renders the phone keyboard correctly 1`] = `
<styled.div>
  <Card>
    <CardSection
      dataTest="keyboard-phone-card"
      expandable={true}
    >
      <CardSectionHeader>
        <Heading
          element="h3"
          type="title3"
        >
          Show/Hide Phone Keyboard 
        </Heading>
      </CardSectionHeader>
      <CardSectionContent>
        <styled.div>
          <Button
            dataTest="keyboard-phone-element-1"
            key="1"
            onClick={[Function]}
            type="secondary"
          >
            1
          </Button>
          <Button
            dataTest="keyboard-phone-element-2"
            key="2"
            onClick={[Function]}
            type="secondary"
          >
            2
          </Button>
          <Button
            dataTest="keyboard-phone-element-3"
            key="3"
            onClick={[Function]}
            type="secondary"
          >
            3
          </Button>
          <Button
            dataTest="keyboard-phone-element-4"
            key="4"
            onClick={[Function]}
            type="secondary"
          >
            4
          </Button>
          <Button
            dataTest="keyboard-phone-element-5"
            key="5"
            onClick={[Function]}
            type="secondary"
          >
            5
          </Button>
          <Button
            dataTest="keyboard-phone-element-6"
            key="6"
            onClick={[Function]}
            type="secondary"
          >
            6
          </Button>
          <Button
            dataTest="keyboard-phone-element-7"
            key="7"
            onClick={[Function]}
            type="secondary"
          >
            7
          </Button>
          <Button
            dataTest="keyboard-phone-element-8"
            key="8"
            onClick={[Function]}
            type="secondary"
          >
            8
          </Button>
          <Button
            dataTest="keyboard-phone-element-9"
            key="9"
            onClick={[Function]}
            type="secondary"
          >
            9
          </Button>
          <Button
            dataTest="keyboard-phone-element-Back"
            key="Back"
            onClick={[Function]}
            type="secondary"
          >
            Back
          </Button>
          <Button
            dataTest="keyboard-phone-element-0"
            key="0"
            onClick={[Function]}
            type="secondary"
          >
            0
          </Button>
          <Button
            dataTest="keyboard-phone-element-Clear"
            key="Clear"
            onClick={[Function]}
            type="secondary"
          >
            Clear
          </Button>
        </styled.div>
      </CardSectionContent>
    </CardSection>
  </Card>
</styled.div>
`;
